src/components/tl/ucp/allgather/allgather_neighbor.c

```diff
@@ -28,11 +28,11 @@ static ucc_rank_t get_recv_from_rank(ucc_rank_t rank, ucc_rank_t size, int i)
         offset_at_step[1] = (-2);
     }
 
-    return (recv_data_from + offset_at_step[i_parity] * ucc_div_round_up(i, 2) +
-            size) %
-           size;
+    return (recv_data_from + offset_at_step[i_parity] * ucc_div_round_up(i, 2) + size) % size;
 }
 
+ucc_status_t ucc_tl_ucp_allgather_neighbor_start(ucc_coll_task_t *coll_task);
```

Comment:
nit: this forward declaration can be avoided if we just permute the function definitions